src/components/checkbox/checkbox.ts

```diff
@@ -39,57 +39,55 @@ export default class IgcCheckboxComponent extends IgcCheckboxBaseComponent {
   @property({ type: Boolean, reflect: true })
   public indeterminate = false;
 
-  private handleClick() {
+  protected handleClick() {
     this.checked = !this.checked;
     this.indeterminate = false;
+    this.emitEvent('igcChange', { detail: this.checked });
```

Comment:
By moving the event emitter here, we will stop emitting when changing the value via keyboard.